raster-foundry/raster-foundry
# Reorganize Tile Server and Instantiate Single Database Pool

## Overview

Does a few things that are related, but somewhat distinct
 - Ensures that only one database pool is instantiated
 - Organizes server similar to API server organization
 - Removes circular dependencies between Stitch, GlobalSummary,
 Mosaic, TileSources, and LayerCache
 - Deletes unused and unnecessary files/functions
 - Adds docstrings to most methods
 - Consolidates some logic
 - Removes most implicit arguments to function calls

### Checklist

~- [ ] Styleguide updated, if necessary~
~- [ ] Swagger specification updated, if necessary~
~- [ ] Symlinks from new migrations present or corrected for any new migrations~

### Demo

![slick-threadpool](https://user-images.githubusercontent.com/898060/28841714-11aa2d16-76c9-11e7-9175-4e8e488e8906.png)

### Notes

A lot of the different parts of the tile server were `object`s, which was convenient, but made it easy to create weird dependencies amongst themselves. It was tough to tell what functions should be organized together. I attempted to separate out the `tool` specific functions, generic project mosaic functions, and more "IO" related functions into their own respective traits. This mimics the API server and hopefully improves discoverability when having to edit files. I'm open to suggestions on improving this further.

There are a few functions with similar functionality that I think warrants looking at how the tile server works again in order to determine if there is more opportunity for simplifying things.

## Testing Instructions

 * Start up servers and `visualvm`, make requests, observe that there is only one thread pool

Closes #2304
